spec/ruby/core/class/fixtures/callback_order.rb

```diff
@@ -0,0 +1,21 @@
+module CoreClassSpecs
+  module Callbacks
+    class Base
+      def self.inherited(subclass)
+        subclass.const_set(:INHERITED_NAME, subclass.name)
+        ORDER << [:inherited, subclass, eval("defined?(#{subclass.name})")]
+        super
+      end
+    end
+
+    ORDER = []
+
+    def self.const_added(const_name)
+      ORDER << [:const_added, const_name, eval("defined?(#{const_name})")]
```

Comment:
I added this `defined?` check, to spec the other change, which is that the constant isn't yet defined when `inherited` is called.

If something causes backward compatibility issues, I think it will be this part. We could totally register the constant before, but the code would be sensibly more finicky, and I understood this was the behavior Matz wanted to try out